cmd/vulcan-exposed-http-resources/main.go

```diff
@@ -0,0 +1,334 @@
+
+func main() {
+	run := func(ctx context.Context, target string, optJSON string, state state.State) (err error) {
+		logger := check.NewCheckLog(checkName)
+		logger = logger.WithFields(logrus.Fields{"target": target, "options": optJSON})
+
+		targetURL, err := url.Parse(target)
+		if err != nil {
+			return err
+		}
+
+		var opt Options
+		if optJSON != "" {
+			if err = json.Unmarshal([]byte(optJSON), &opt); err != nil {
+				return err
+			}
+		}
+
+		resources := opt.Resources
+		if len(resources) < 1 {
+			defaultResourcesFile, err := ioutil.ReadFile("resources.yaml")
+			if err != nil {
+				return err
+			}
+			err = yaml.Unmarshal(defaultResourcesFile, &resources)
+			if err != nil {
+				return err
+			}
+		}
+
+		vuln, checked, err := exposedResources(logger, targetURL, resources)
+		if err != nil {
+			return err
+		}
+
+		if vuln != nil {
+			if len(vuln.Resources) > 0 {
+				err = filterFalsePositives(vuln, checked)
+				if err != nil {
+					return err
+				}
+			}
+			state.AddVulnerabilities(*vuln)
+		}
+
+		return nil
+	}
+
+	c := check.NewCheckFromHandler(checkName, run)
+	c.RunAndServe()
+}
+
+func exposedResources(l *logrus.Entry, targetURL *url.URL, httpResources []Resource) (*report.Vulnerability, map[string]int, error) {
+	var vuln *report.Vulnerability
+	vulnResources := []map[string]string{}
+	checkedResources := map[string]int{}
+
+	for _, httpResource := range httpResources {
+		checkedResources[rankConfidence(httpResource)]++
+
+		foundResource, err := checkResource(l, targetURL, httpResource)
+		if err != nil {
+			return nil, nil, err
+		}
+		if foundResource != nil {
+			vulnResources = append(vulnResources, foundResource)
+		}
+	}
+
+	if len(vulnResources) > 0 {
+		vuln = &exposedVuln
+		vuln.Resources[0].Rows = vulnResources
+	}
+
+	return vuln, checkedResources, nil
+}
+
+func checkResource(l *logrus.Entry, targetURL *url.URL, httpResource Resource) (map[string]string, error) {
+	foundPathsChan := make(chan string)
+
+	go func() {
+		limiter := rate.NewLimiter(rate.Limit(rateLimit), burst)
+		var wg sync.WaitGroup
+		for _, p := range httpResource.Paths {
+			wg.Add(1)
+			go func() {
+				err := limiter.Wait(context.Background())
+				if err != nil {
+					l.Error(err)
+					return
+				}
+
+				targetResource, err := url.Parse(targetURL.String())
+				if err != nil {
+					l.Error(err)
+					return
+				}
+
+				targetResource.Path = path.Join(targetResource.Path, p)
+				if strings.HasSuffix(p, "/") && !strings.HasSuffix(targetResource.Path, "/") {
+					targetResource.Path += "/"
+				}
+
+				positive, err := checkPath(l, targetResource, httpResource)
+				if err != nil {
+					l.Error(err)
+					return
+				}
+
+				if positive {
+					foundPathsChan <- targetResource.String()
+				}
+
+				wg.Done()
+			}()
+		}
+		wg.Wait()
+		close(foundPathsChan)
+	}()
+
+	foundPaths := []string{}
+	for foundPath := range foundPathsChan {
+		foundPaths = append(foundPaths, foundPath)
+	}
+
+	if len(foundPaths) > 0 {
```

Comment:
Also not a strong opinion, but here, I would do:
```
if len(foundPaths) < 1 {
    return nil, nil
}
var severityRank string
		if httpResource.Severity != nil {
			if *httpResource.Severity > exposedVuln.Score {
				exposedVuln.Score = *httpResource.Severity
			}
			severityRank = rankSeverity(*httpResource.Severity)
		} else {
			severityRank = rankSeverity(report.SeverityThresholdHigh)
		}

		return map[string]string{
			"Score":       fmt.Sprintf("%.01f", *httpResource.Severity),
			"Severity":    severityRank,
			"Confidence":  rankConfidence(httpResource),
			"Description": httpResource.Description,
			"URL":         strings.Join(foundPaths, "\n"),
		}, nil
```

Reply:
I agree! Fixed :+1: